File: Udemy/algorithms/fizzBuzz.js
// O(n)
function fizzBuzz(num) {

  for(var i=0; i<num; i++) {
    var res = "";
    if(i % 3 == 0){
      res += "Fizz";
    }
    if(i % 5 ==0 ){
      res += "Buzz";
    }
    if(res=="")
      console.log(i);
    else
      console.log(res);
  }

}

fizzBuzz(20);


// or
function fizzBuzz_v1(num) {
  for (var i = 1; i <= num; i++) {
    if (i % 15 === 0) console.log('FizzBuzz');
    else if (i % 3 === 0) console.log('Fizz');
    else if (i % 5 === 0) console.log('Buzz');
    else console.log(i);
  }
}

File: CodingTest/CodingTest/FizzBuzz.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CodingTest
{
    class FizzBuzz
    {
        public void FizzBuz()
        {
            List<int> collection = new List<int>() { 2, 3, 5, 7 };

            foreach (int x in collection)
            {
                if (x % 3 == 0 && x % 5 == 0)
                {
                    Console.WriteLine("FizzBuzz");
                }
                else if (x % 3 == 0)
                {
                    Console.WriteLine("Fizz");
                }
                else if (x % 5 == 0)
                {
                    Console.WriteLine("Buzz");
                }
            }


            collection = new List<int>() { 2, 3, 5, 7 };

            foreach (int x in collection)
            {
                if (x % 3 == 0)
                {
                    Console.Write("Fizz");
                }
                if (x % 5 == 0)
                {
                    Console.Write("Buzz");
                }
            }

            collection = new List<int>() { 2, 3, 5, 7 };

            foreach (int x in collection)
            {
                bool insertNewline = false;
                if (x % 3 == 0)
                {
                    Console.Write("Fizz");
                    insertNewline = true;
                }
                if (x % 5 == 0)
                {
                    Console.Write("Buzz");
                    insertNewline = true;
                }

                if (insertNewline)
                {
                    Console.Write(Environment.NewLine);
                }
            }


        }
    }
}
